geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/SqlHandler.java

```diff
@@ -23,34 +23,51 @@
 import java.util.Date;
 import java.util.List;
 
+import javax.sql.DataSource;
+
 import org.apache.geode.InternalGemFireException;
 import org.apache.geode.annotations.Experimental;
 import org.apache.geode.cache.Operation;
 import org.apache.geode.cache.Region;
 import org.apache.geode.connectors.jdbc.JdbcConnectorException;
 import org.apache.geode.connectors.jdbc.internal.configuration.ConnectorService;
 import org.apache.geode.internal.cache.InternalCache;
+import org.apache.geode.internal.jndi.JNDIInvoker;
 import org.apache.geode.pdx.PdxInstance;
 
 @Experimental
 public class SqlHandler {
   private final JdbcConnectorService configService;
-  private final DataSourceManager manager;
   private final TableMetaDataManager tableMetaDataManager;
+  private final DataSourceFactory dataSourceFactory;
 
-  public SqlHandler(DataSourceManager manager, TableMetaDataManager tableMetaDataManager,
-      JdbcConnectorService configService) {
-    this.manager = manager;
+  public SqlHandler(TableMetaDataManager tableMetaDataManager, JdbcConnectorService configService,
+      DataSourceFactory dataSourceFactory) {
     this.tableMetaDataManager = tableMetaDataManager;
     this.configService = configService;
+    this.dataSourceFactory = dataSourceFactory;
+  }
+
+  public SqlHandler(TableMetaDataManager tableMetaDataManager, JdbcConnectorService configService) {
+    this(tableMetaDataManager, configService,
+        dataSourceName -> JNDIInvoker.getDataSource(dataSourceName));
   }
 
-  public void close() {
-    manager.close();
+  public interface DataSourceFactory {
```

Comment:
I would sink this inner-interface to the bottom of the outer-class.

Reply:
will move it to the end of the class